src/CoreWCF.WebHttp/src/CoreWCF/Dispatcher/SingleBodyParameterMessageFormatter.cs

```diff
@@ -0,0 +1,387 @@
+            {
+                return new SingleBodyParameterXmlSerializerMessageFormatter(operation, type, isRequestFormatter, xsob, xmlSerializerManager);
+            }
+
+            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException(SR.Format(SR.OnlyDataContractAndXmlSerializerTypesInUnWrappedMode, operation.Name)));
+        }
+
+        public static SingleBodyParameterMessageFormatter CreateJsonFormatter(OperationDescription operation, Type type, bool isRequestFormatter)
+        {
+            DataContractSerializerOperationBehavior dcsob = ((KeyedByTypeCollection<IOperationBehavior>)operation.OperationBehaviors).Find<DataContractSerializerOperationBehavior>();
+            if (dcsob == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.JsonFormatRequiresDataContract, operation.Name, operation.DeclaringContract.Name, operation.DeclaringContract.Namespace)));
+            }
+
+            return new SingleBodyParameterDataContractMessageFormatter(operation, type, isRequestFormatter, true, dcsob);
+        }
+
+        private BodyWriter CreateBodyWriter(object body)
+        {
+            XmlObjectSerializer serializer;
+            if (body != null)
+            {
+                serializer = GetOutputSerializer(body.GetType());
+                if (serializer == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException(SR.Format(SR.CannotSerializeType, body.GetType(), OperationName, ContractName, ContractNs, _serializerType)));
+                }
+            }
+            else
+            {
+                serializer = null;
+            }
+
+            return new SingleParameterBodyWriter(body, serializer);
+        }
+
+        protected virtual object ReadObject(Message message)
+        {
+            if (HttpStreamFormatter.IsEmptyMessage(message))
+            {
+                return null;
+            }
+
+            XmlObjectSerializer[] inputSerializers = GetInputSerializers();
+            XmlDictionaryReader reader = message.GetReaderAtBodyContents();
+            if (inputSerializers != null)
+            {
+                for (int i = 0; i < inputSerializers.Length; ++i)
+                {
+                    if (inputSerializers[i].IsStartObject(reader))
+                    {
+                        return inputSerializers[i].ReadObject(reader, false);
+                    }
+                }
+            }
+
+            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SerializationException(SR.Format(SR.CannotDeserializeBody, reader.LocalName, reader.NamespaceURI, OperationName, ContractName, ContractNs, _serializerType)));
+        }
+
+        internal class NullMessageFormatter : IDispatchMessageFormatter, IClientMessageFormatter
+        {
+            // TODO: Can I remove this?
+
```

Comment:
Yes, this is for JSONP support which we aren't implementing

Reply:
This has been addressed.